test/unit/webapps/test_response_wrapper.py

```diff
@@ -0,0 +1,49 @@
+from unittest.mock import Mock
+
+from fastapi.applications import FastAPI
+from fastapi.param_functions import Depends
+from fastapi.testclient import TestClient
+from starlette.responses import Response
+
+from galaxy.app_unittest_utils.galaxy_mock import MockApp
+from galaxy.webapps.galaxy.api import GalaxyASGIResponse
+from galaxy.work.context import SessionRequestContext
+
+app = FastAPI()
+
+client = TestClient(app)
+
+
+def get_trans(response: Response) -> SessionRequestContext:
+    galaxy_response = GalaxyASGIResponse(response)
+    return SessionRequestContext(
+        app=MockApp(),
+        request=Mock(),
+        response=galaxy_response,
+    )
+
+
+@app.get("/test/change_headers")
+async def change_headers(trans=Depends(get_trans)):
+    trans.response.headers["test-header"] = "test-value"
+
+
+@app.get("/test/change_content_type")
+async def change_content_type(trans=Depends(get_trans)):
+    trans.response.set_content_type("test-content-type")
```

Comment:
Does this work, as temporary workaround ?
```suggestion
async def change_content_type(trans=Depends(get_trans), response: Response):
    trans.response.__response = response
    trans.response.set_content_type("test-content-type")
```
If I understand the linked code correctly the problem is that the response is in a subresponse, which is everything that comes in via dependencies ? So maybe the trick works ?

Reply:
I'm afraid it doesn't work :disappointed: 
This new response is probably also a subresponse.